### PR TITLE
Fix unhandled exception in `find_bins_in_path`

### LINKED ISSUE #51: Helper function `find_bins_in_path` can raise unhandled exception
When using the helper function `find_bins_in_path` on linux, if an item exists in your PATH variable that does not exist on disk, an exception will be raised and _all tab completion will be broken_, not just bin completion.

### Steps to Reproduce

```bash
# run demo.py with nonexistent item on PATH
PATH="$PATH:nonexistent" python3 demo.py
# ...
[19:50:51] pypsi )> mac<tab>  # observe pressing tab here does nothing
```

### PR DESCRIPTION
**Pull request by Rudedog9d**
Closes #51 

Fixes an unhandled exception in helper function `find_bins_in_path`. Covers 3 main cases:

- A path in PATH is not found (`FileNotFoundError`)
- A path in PATH is not readable (`PermssionError`)
- Path is `unset` for some reason

### CODE REVIEW

File: test/test_completers/test_find_bins_in_path.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+        self.environ_patch = patch('os.environ')
+        self.environ_mock = self.environ_patch.start()
+
+    def teardown(self):
+        self.environ_mock.stop()
+
+    def test_path_unset(self):
+        # An empty or missing PATH will result in checking `./`, which checks the current repo
+        self.environ_mock.return_value = f"./"
```
Comment (ameily) [inline test/test_completers/test_find_bins_in_path.py]: The CI tests run on pre-Python 3.7. Can you update to remove f-strings and use `%` or `.format()` instead?

Comment (Rudedog9d) [reply test/test_completers/test_find_bins_in_path.py]: I just pushed some simplified unit tests that don't use f-strings (the old unit tests didn't actually test anything!). I'll make sure there aren't any other newer features in use.

**Comment by ameily** [comment]
@Rudedog9d Can you merge in upstream `master`? I migrated the project over to travis-ci.com, which fixed build issues but it looks like this PR won't build now.

**Comment by Rudedog9d** [comment]
@ameily Hmm, looks like it still failed :( Let me know if you want me to try something else

**Review by ameily — APPROVED** [review]

**Comment by ameily** [comment]
Idk why but travis-ci is failing because `coverage combine` is breaking. I found a bunch of posts on Github and StackOverflow, with no clear answer. People seem to think the issue is tied to running coverage inside a docker container. I'm going to merge this.

**Comment by ameily** [comment]
@Rudedog9d It's merged, thanks for the contribution! The problem with travis-ci appears to be that the unit tests weren't unpatching some methods. I updated the unit tests on master and everything appears to be working.

**Comment by Rudedog9d** [comment]
@ameily Happy to help! Glad you got Travis figured out haha

**Pull request merged**